src/query/runner/oprf_ipa.rs

```diff
@@ -0,0 +1,127 @@
+use std::marker::PhantomData;
+
+use futures::{Stream, TryStreamExt};
+
+use crate::{
+    error::Error,
+    ff::{Field, Gf2, PrimeField, Serializable},
+    helpers::{
+        query::{IpaQueryConfig, QuerySize},
+        BodyStream, RecordsStream,
+    },
+    protocol::{
+        basics::ShareKnownValue,
+        context::{UpgradableContext, UpgradedContext},
+        prf_sharding::{attribution_and_capping_and_aggregation, PrfShardedIpaInputRow},
+        BreakdownKey, Timestamp, TriggerValue,
+    },
+    report::{EventType, OprfReport},
+    secret_sharing::{
+        replicated::{malicious::ExtendableField, semi_honest::AdditiveShare as Replicated},
+        SharedValue,
+    },
+};
+
+pub struct OprfIpaQuery<C, F> {
+    config: IpaQueryConfig,
+    phantom_data: PhantomData<(C, F)>,
+}
+
+impl<C, F> OprfIpaQuery<C, F> {
+    pub fn new(config: IpaQueryConfig) -> Self {
+        Self {
+            config,
+            phantom_data: PhantomData,
+        }
+    }
+}
+
+impl<C, F> OprfIpaQuery<C, F>
+where
+    C: UpgradableContext,
+    C::UpgradedContext<F>: UpgradedContext<F, Share = Replicated<F>>,
+    C::UpgradedContext<Gf2>: UpgradedContext<Gf2, Share = Replicated<Gf2>>,
+    F: PrimeField + ExtendableField,
+    Replicated<F>: Serializable + ShareKnownValue<C, F>,
+    Replicated<Gf2>: Serializable + ShareKnownValue<C, Gf2>,
+{
+    #[tracing::instrument("oprf_ipa_query", skip_all, fields(sz=%query_size))]
+    pub async fn execute<'a>(
+        self,
+        ctx: C,
+        query_size: QuerySize,
+        input_stream: BodyStream,
+    ) -> Result<Vec<Replicated<F>>, Error> {
+        let Self {
+            config,
+            phantom_data: _,
+        } = self;
+        tracing::info!("New query: {config:?}");
+        let sz = usize::from(query_size);
+
+        let input = if config.plaintext_match_keys {
+            let mut v = assert_stream_send(RecordsStream::<
```

Comment:
Do we absolutely need this "assert_stream_send" call? Can you try to remove that and see if it compiles?